### PR TITLE
add TealVerify function

### PR DESCRIPTION
**Pull request by ori-shem-tov**
Adding `TealVerify`, a function which verifies signatures generated by `TealSign` and `TealSignFromProgram`.
Helpful for testing and debugging.

### CODE REVIEW

**Comment by jasonpaulos** [comment]
Thanks for making this PR, this seems like a good addition.

It would be great if you could also do the following to test the new `TealVerify` function:
* Change the existing test code to use `TealVerify`: https://github.com/algorand/go-algorand-sdk/blob/af44e0eeb717218cbb661b182b2abf96238c5f50/crypto/crypto_test.go#L725-L728
* In that same test, modify a single byte of `data` and call `TealVerify` again to make sure it fails.

**Review by jasonpaulos — COMMENTED** [review]
Thank you for adding the test. This looks pretty good to me, though I do have one concern about the signature argument type.

File: crypto/crypto.go [hunk 1]
```diff
@@ -697,3 +697,11 @@ func TealSignFromProgram(sk ed25519.PrivateKey, data []byte, program []byte) (ra
 	addr := AddressFromProgram(program)
 	return TealSign(sk, data, addr)
 }
+
+// TealVerify verifies signatures generated by TealSign and TealSignFromProgram
+func TealVerify(pk ed25519.PublicKey, data []byte, contractAddress types.Address, rawSig []byte) bool {
```
[the remark this answers is not in the document]
Comment (ori-shem-tov) [reply crypto/crypto.go:702]: I guess it binds it with `TealSign` and `TealSignFromProgram` which makes sense and also adds readability.
I've created a commit for that, let me know if you think I should push it.

[the remark this answers is not in the document]
Comment (winder) [reply crypto/crypto.go:702]: Good idea, I think it makes sense to use `types.Signature`. Yes please make that change @ori-shem-tov

**Review by jasonpaulos — APPROVED** [review]
Thank you for quickly addressing our feedback @ori-shem-tov, this looks good to merge

**Pull request merged**